### ISSUE TITLE
crash on line typing (TypeError: 'NoneType' object is not iterable)

### ISSUE THREAD

**Issue by nil0x42** [bpython/bpython#726]
## Steps to reproduce:
```python
bpython version 0.16 on top of Python 3.6.4 /usr/bin/python
>>> from PyQt5 import Qt
>>> Qt.QWidget.setFocusPolicy(Qt.
```
(error happens after typing last '.' dot)

## Traceback:
```
Traceback (most recent call last):
  File "/usr/bin/bpython", line 11, in <module>
    load_entry_point('bpython==0.16', 'console_scripts', 'bpython')()
  File "/usr/lib/python3.6/site-packages/bpython/curtsies.py", line 190, in main
    exit_value = repl.mainloop()
  File "/usr/lib/python3.6/site-packages/bpython/curtsies.py", line 125, in mainloop
    self.process_event_and_paint(e)
  File "/usr/lib/python3.6/site-packages/bpython/curtsies.py", line 89, in process_event_and_paint
    self.process_event(e)
  File "/usr/lib/python3.6/site-packages/bpython/curtsiesfrontend/repl.py", line 584, in process_event
    return self.process_key_event(e)
  File "/usr/lib/python3.6/site-packages/bpython/curtsiesfrontend/repl.py", line 736, in process_key_event
    self.add_normal_character(e)
  File "/usr/lib/python3.6/site-packages/bpython/curtsiesfrontend/repl.py", line 972, in add_normal_character
    self.cursor_offset += 1
  File "/usr/lib/python3.6/site-packages/bpython/curtsiesfrontend/repl.py", line 1544, in _set_cursor_offset
    self.update_completion()
  File "/usr/lib/python3.6/site-packages/bpython/curtsiesfrontend/repl.py", line 1003, in update_completion
    self.list_win_visible = BpythonRepl.complete(self, tab)
  File "/usr/lib/python3.6/site-packages/bpython/repl.py", line 720, in complete
    self.current_line, matches, completer)
  File "/usr/lib/python3.6/site-packages/bpython/repl.py", line 331, in update
    self.orig_cursor_offset, self.orig_line)
TypeError: 'NoneType' object is not iterable
```

## System:
Archlinux x64
bpython 0.16
python 3.6.4

**Comment by AustinTSchaffer** [comment]
I'm getting a similar issue, but without requiring an external dependency. Figured I'd add my own reproduction steps. The issue is happening for me when I try to tab-complete a dict key that contains a slash. I've also included a screenshot which shows color information.

The issue for me seems to be any time I try to tab-complete a dict key when all of the keys contain a slash and have the same prefix. This is occurring for me in cases where I'm using the names of a collection of files as keys in a dict. My workaround is to not use tab completion in those cases.

I love this project and this tool by the way, use it every day. Let me know if there's anything else I can do to help with this.

## Steps to Reproduce

For reference, the last sequence of keypresses was `x`, `[`, then `<tab>`

```
bpython version 0.18 on top of Python 3.5.3 /home/austin/workspace/migrations/xxx/venv/bin/python3
>>> x = {}
>>> x["a/b"] = 1
>>> x["a/c"] = 1
>>> x["a/c"] = 1
Traceback (most recent call last):
  File "/home/austin/workspace/migrations/xxx/venv/bin/bpython", line 10, in <module>
    sys.exit(main())
  File "/home/austin/workspace/migrations/xxx/venv/lib/python3.5/site-packages/bpython/curtsies.py", line 191, in main
    exit_value = repl.mainloop(True, paste)
  File "/home/austin/workspace/migrations/xxx/venv/lib/python3.5/site-packages/bpython/curtsies.py", line 126, in mainloop
    self.process_event_and_paint(e)
  File "/home/austin/workspace/migrations/xxx/venv/lib/python3.5/site-packages/bpython/curtsies.py", line 90, in process_event_and_paint
    self.process_event(e)
  File "/home/austin/workspace/migrations/xxx/venv/lib/python3.5/site-packages/bpython/curtsiesfrontend/repl.py", line 595, in process_event
    return self.process_key_event(e)
  File "/home/austin/workspace/migrations/xxx/venv/lib/python3.5/site-packages/bpython/curtsiesfrontend/repl.py", line 726, in process_key_event
    self.on_tab()
  File "/home/austin/workspace/migrations/xxx/venv/lib/python3.5/site-packages/bpython/curtsiesfrontend/repl.py", line 841, in on_tab
    cursor_and_line = self.matches_iter.substitute_cseq()
  File "/home/austin/workspace/migrations/xxx/venv/lib/python3.5/site-packages/bpython/repl.py", line 333, in substitute_cseq
    self.completer)
  File "/home/austin/workspace/migrations/xxx/venv/lib/python3.5/site-packages/bpython/repl.py", line 353, in update
    self.orig_cursor_offset, self.orig_line)
TypeError: 'NoneType' object is not iterable
```

![image](https://user-images.githubusercontent.com/16184219/68888706-72b78d80-06e9-11ea-80dd-fa21b64de544.png)

## Working

```python
x = {}
x["a/b"] = 1
x[
```

```python
x = {}
x["a/b"] = 1
x["a.b"] = 1
x[
```

```python
x = {}
x["a/b"] = 1
x["a.b"] = 1
x["a/c"] = 1
x[
```

```python
x = {}
x["a/b"] = 1
x["b/d"] = 1
x[
```

```python
x = {}
x["a/"] = 1
x["b/"] = 1
x["a/b"] = 1
x[
```

## Not Working

```python
x = {}
x["a/"] = 1
x["a//"] = 1
x[
```

```python
x = {}
x["./a"] = 1
x["./b"] = 1
x["./c"] = 1
x[
```

```python
x = {}
x["asdfasdfasdf/1"] = 1
x["asdfasdfasdf/2"] = 2
x["asdfasdfasdf/3"] = 3
x[
```

## System

Debian 9 (`uname -a`: Linux debian 4.9.0-9-amd64 #1 SMP Debian 4.9.168-1+deb9u5 (2019-08-11) x86_64 GNU/Linux)
bpython 0.18
Python 3.5.3

**Comment by sebastinas** [comment]
Fixed in 34f0e700082bf79223689ff120dddf4e46442129.
